src/Tool/BearerAuthorizationTrait.php

```diff
@@ -9,7 +11,7 @@
  */
 trait BearerAuthorizationTrait
 {
-    protected function getAuthorizationHeaders($token = null)
+    protected function getAuthorizationHeaders(AccessToken $token = null)
     {
         return ['Authorization' => 'Bearer ' . $token];
```

Comment:
This is exactly what I mean... the usage of the token here means that we explicitly **do** want to allow strings instead of instances.